tests/test_conservation.py

```diff
@@ -0,0 +1,377 @@
+import climt
+from sympl import (
+    get_constant, AdamsBashforth,
+    TendencyComponent, TendencyStepper)
+from datetime import timedelta
+import numpy as np
+
+
+class Conservation(object):
+
+    def get_model_state(self, component):
+
+        state = climt.get_default_state([component])
+        return self.modify_state(state)
+
+    def modify_state(self, state):
+        return state
+
+    def get_component_instance(self):
+        pass
+
+    def get_quantity_forcing(self, state):
+        pass
+
+    def get_quantity_amount(self, state):
+        pass
+
+    def get_steppable_component(self):
+
+        component = self.get_component_instance()
+
+        if isinstance(component, TendencyComponent):
+            return AdamsBashforth(component)
+        else:
+            return component
+
+    def get_new_state_and_diagnostics(self, state,
+                                      component, time_step):
+
+        if isinstance(component, TendencyStepper):
+            diag, state = component(state, time_step)
+            state.update(diag)
+
+            return state
+        else:
+            diag, new_state = component(state, time_step)
+            state.update(new_state)
+            state.update(diag)
+
+            return state
+
+    def test_quantity_is_conserved(self):
+
+        component = self.get_steppable_component()
+        state = self.get_model_state(component)
+        time_step = timedelta(seconds=1)
+
+        first_state = self.get_new_state_and_diagnostics(state,
+                                                         component, time_step)
+
+        old_amount = self.get_quantity_amount(first_state)
+
+        new_state = self.get_new_state_and_diagnostics(first_state,
+                                                       component, time_step)
+
+        new_amount = self.get_quantity_amount(new_state)
+
+        forcing_amount = self.get_quantity_forcing(new_state)*time_step.total_seconds()
+
+        assert np.isclose(new_amount - old_amount, forcing_amount,
+                          rtol=0, atol=1e-3)
+
+
+def get_pressure_thickness(state):
+    return (state['air_pressure_on_interface_levels'][:-1] -
+            state['air_pressure_on_interface_levels'][1:]).rename(
+                dict(interface_levels='mid_levels'))
+
+
+def vertical_integral(state, quantity):
+    g = get_constant('gravitational_acceleration', 'm/s^2')
+    dp = get_pressure_thickness(state)
+
+    return (quantity*dp/g).sum().values
+
+
+def get_moist_enthalpy(state):
+    Cpd = get_constant('heat_capacity_of_dry_air_at_constant_pressure', 'J/kg/degK')
+    Lv = get_constant('latent_heat_of_condensation', 'J/kg')
+
+    dry_enthalpy = vertical_integral(state, Cpd*state['air_temperature'])
+
+    moisture_enthalpy = 0.0
+    if 'specific_humidity' in state:
+        moisture_enthalpy = vertical_integral(state, Lv*state['specific_humidity'])
```

Comment:
You should convert `state['specific_humidity']` and any other state variables (like `air_temperature` above) to the units you want them to be here before using their values. This avoids errors if you apply this test to a component that uses non-SI units.